loader/include/Geode/loader/ModMetadata.hpp

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include "../utils/Result.hpp"
+#include "../utils/VersionInfo.hpp"
+#include "ModInfo.hpp"
+#include "Setting.hpp"
+#include "Types.hpp"
+
+#include <json.hpp>
+#include <memory>
+
+namespace geode {
+    namespace utils::file {
+        class Unzip;
+    }
+
+    struct GEODE_DLL [[deprecated("use ModMetadata::Dependency instead")]] Dependency;
+    struct [[deprecated("use ModMetadata::IssuesInfo instead")]] IssuesInfo;
+
+    class ModMetadataImpl;
+
+    /**
+     * Represents all the data gather-able
+     * from mod.json
+     */
+    class GEODE_DLL ModMetadata {
+        class Impl;
+        std::unique_ptr<Impl> m_impl;
+
+    public:
+        ModMetadata();
+        explicit ModMetadata(std::string id);
+        ModMetadata(ModMetadata const& other);
+        ModMetadata(ModMetadata&& other) noexcept;
+        ModMetadata& operator=(ModMetadata const& other);
+        ModMetadata& operator=(ModMetadata&& other) noexcept;
+        ~ModMetadata();
+
+        struct GEODE_DLL Dependency {
+            enum class Importance : uint8_t { Required, Recommended, Suggested };
+            std::string id;
+            ComparableVersionInfo version;
+            Importance importance = Importance::Required;
+            Mod* mod = nullptr;
+            [[nodiscard]] bool isResolved() const;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma ide diagnostic ignored "google-explicit-constructor"
+            operator geode::Dependency();
+            operator geode::Dependency() const;
+
+            static Dependency fromDeprecated(geode::Dependency const& value);
+#pragma clang diagnostic pop
+        };
+
+        struct GEODE_DLL Incompatibility {
+            std::string id;
+            ComparableVersionInfo version;
+            Mod* mod = nullptr;
+            [[nodiscard]] bool isResolved() const;
+        };
+
+        struct IssuesInfo {
+            std::string info;
+            std::optional<std::string> url;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma ide diagnostic ignored "google-explicit-constructor"
+            operator geode::IssuesInfo();
+            operator geode::IssuesInfo() const;
+
+            static IssuesInfo fromDeprecated(geode::IssuesInfo const& value);
+#pragma clang diagnostic pop
+        };
+
+        /**
+         * Path to the mod file
+         */
+        [[maybe_unused, nodiscard]] ghc::filesystem::path getPath() const;
```

Comment:
whats the point of maybe_unused along with nodiscard, what does maybe_unused even do to a method

Reply:
maybe_unused silences unused warnings

Reply:
but why does it even consider it unused, the whole class is exported

Reply:
idk, ask clang-tidy, i can remove the attributes if you dont like them